src/cart/SConscript

```diff
@@ -22,6 +22,25 @@ SRC = ['crt_bulk.c', 'crt_context.c', 'crt_corpc.c',
        'crt_swim.c', 'crt_tree.c', 'crt_tree_flat.c', 'crt_tree_kary.c',
        'crt_tree_knomial.c', 'crt_hlc.c', 'crt_hlct.c']
 
+LIB_SRC = ['crt_utils.c']
+
+##############################
+# Create cart utility library
+##############################
+def build_utility_library():
+    """Build cart utility library"""
+
+    Import('env', 'prereqs')
+    denv = env.Clone()
+    denv.AppendUnique(CPPPATH=[Dir('.').srcnode()])
+    denv.AppendUnique(CPPPATH=[Dir('include').srcnode()])
+
+    prereqs.require(denv, 'mercury', headers_only=True)
+    cart_utils_lib  = daos_build.library(denv, 'cart_utils', LIB_SRC)
```

Comment:
Our other libraries are standalone.   If you only use cart APIs, you only will need -lcart on the command line.   Because you don't add any LIBS= here, this is not the case.

From what I see you need LIBS=['cart']

Reply:
The purpose of the crt_utilis library is to have an frame work for testing of the cart api and functionality.  Normally, it should have just been a local library used in internal testing.  However, it was used in /utils/ctl and /utils/self_test and we cannot divorce ourselves from it at this time.  We do not want to add it to libcart because we do want to make it available for normal usage.  Code that uses this library has set LIBS=['cart_utils'], this include ctl, self_test and the cart test programs.